### PR TITLE
[RUN-2879] add a hidden WebContents + WebUI for hosting our JS

### PR DESCRIPTION
**Pull request by toshok**
The bulk of this is boilerplate needed to get another resource pak built.  A new URL `chrome://recordreplay/` is added, which instantiates a RecordReplayUI, which serves requests from the html/js files that are included in the resource pak.

The html/typescript files are:

* `chrome/browser/resources/record_replay/record_replay.html`
* `chrome/browser/resources/record_replay/record_replay.ts`

The build system has support for multiple typescript files being compiled/bundled together as a single .js file, so we can be as file-happy as we want.

The html probably won't need modifiying - it's not displayed anywhere, and it just loads the bundled JS.

### CODE REVIEW

File: chrome/browser/ui/webui/record_replay/record_replay_ui.cc [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+RecordReplayUI::RecordReplayUI(
+    content::WebUI* web_ui)
+    : ui::MojoWebUIController(web_ui) {
+  content::WebUIDataSource* data_source =
+      content::WebUIDataSource::CreateAndAdd(
+          web_ui->GetWebContents()->GetBrowserContext(),
+          chrome::kChromeUIRecordReplayHost);
+
+  data_source->AddResourcePaths(base::make_span(
+      kRecordReplayResources, kRecordReplayResourcesSize));
```
Comment (toshok) [inline chrome/browser/ui/webui/record_replay/record_replay_ui.cc:21]: this call maps the path part of request urls (the html/js file names) to the corresponding generated resource IDs

File: chrome/browser/ui/webui/record_replay/record_replay_ui.cc [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  data_source->AddResourcePath("", IDR_RECORD_REPLAY_RECORD_REPLAY_HTML);
```
Comment (toshok) [inline chrome/browser/ui/webui/record_replay/record_replay_ui.cc:22]: This call registers the empty path, mapping it to our HTML.  I'm not 100% sure this is necessary, but it feels like it is, given the url we're loading is `chrome://recordreplay/`, without a file.

File: chrome/browser/resources/record_replay/record_replay.gni [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# Copyright 2022 The Chromium Authors
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+# List of files that don't need to be passed to html_to_wrapper().
+non_web_component_files =
+    [
+      "record_replay.ts",
```
Comment (toshok) [inline chrome/browser/resources/record_replay/record_replay.gni:8]: this is where we'd add additional `.ts` files to include them in the `js` resource bundle.

File: chrome/browser/ui/browser.cc [hunk 4]
```diff
@@ -563,6 +563,25 @@ Browser::Browser(const CreateParams& params)
...
+  recordreplay_params.is_never_visible = true;
+
+  recordreplay_contents_ = content::WebContents::Create(recordreplay_params);
+  recordreplay_contents_->SetDelegate(this);
+  recordreplay_contents_->GetController().LoadURL(
+    GURL(base::StringPiece(chrome::kChromeUIRecordReplayPageURL)),
+    content::Referrer(),
+    ui::PAGE_TRANSITION_TYPED, // wrong.  the user didn't type this
+    std::string()
+  );
```
Comment (toshok) [inline chrome/browser/ui/browser.cc:583]: for lack of a better spot, this is where I created the hidden `WebContents` and cause it to load our chrome:// url.  afaict, we can put it anywhere we have a browser (to set the delegate) and profile, so likely we could host this code closer to where the button is created.

Comment (toshok) [reply chrome/browser/ui/browser.cc]: Moved this to the toolbar button

**Review by kannanvijayan — APPROVED** [review]

**Pull request merged**